src/test/java/cz/datadriven/utils/config/view/SerializationTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Datadriven.cz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.datadriven.utils.config.view;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValueFactory;
+import cz.datadriven.utils.config.view.annotation.ConfigView;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.NotSerializableException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class SerializationTest {
+
+  @ConfigView
+  interface TestConfigViewSerializableInterface extends Serializable {
```

Comment:
We could add `Serializable` interface by default in `ConfigViewFactory`, at least for interfaces using `ConfigView` annotation.

```java
enhancer.setInterfaces(new Class[] {configViewClass, Serializable.class});
```
Explicit declaration of `Serializable` wouldn't be necessary then.

Reply:
I think we should always make this explicit, because of the possible memory overhead this can introduce after deserialization.

Reply:
agree